mlflow/types/utils.py

```diff
@@ -286,3 +289,16 @@ def _is_spark_df(x) -> bool:
         return isinstance(x, pyspark.sql.dataframe.DataFrame)
     except ImportError:
         return False
+
+
+def _infer_schema_from_type_hint(type_hint, example=None):
+    if type_hint == List[str]:
+        return Schema([ColSpec(type="string", name=None)])
+    elif type_hint == List[Dict[str, str]]:
+        if example:
+            return Schema([ColSpec(type="string", name=name) for name in example[0]])
```

Comment:
Do we want the name to be the key of the dict?

Reply:
Also, if the example contains more than 1 dictionary within the list, should we validate that all keys match among all dictionaries?

Reply:
> Do we want the name to be the key of the dict?

`name` represents the dict key :)

> Also, if the example contains more than 1 dictionary within the list, should we validate that all keys match among all dictionaries?

ok, will add that validation